### ISSUE TITLE
Step reformatting of "pbxproj" file is breaking ionic-archive step

### ISSUE THREAD

**Issue by patrickbussmann** [bitrise-steplib/steps-ios-auto-provision-appstoreconnect#34]
I have this workflow snippet.

```yaml
    - npm@1:
        inputs:
        - command: install
    - script@1:
        inputs:
        - content: ionic cordova platform add ios --no-resources
    - ionic-prepare@1:
        inputs:
        - platform: ios
    - recreate-user-schemes@1:
        inputs:
        - project_path: "$BITRISE_SOURCE_DIR/platforms/ios/App.xcodeproj"
    - ios-auto-provision-appstoreconnect@0:
        inputs:
        - project_path: "$BITRISE_SOURCE_DIR/platforms/ios/App.xcworkspace"
        - scheme: App
        - distribution_type: "$BITRISE_EXPORT_METHOD"
    - ionic-archive@2:
        inputs:
        - ionic_version: latest
        - cordova_version: latest
        - options: --prod -- --buildFlag="-UseModernBuildSystem=0"
        - run_ionic_prepare: 'false'
        - platform: "$IONIC_PLATFORM"
```

After the `recreate-user-schemes` step the `platforms/ios/App.xcodeproj/project.pbxproj` file looks normal. Like this:

<img width="555" alt="2021-01-03 00_45_11-S - build #744 (d4436f25b3b693b3) - Bitrise" src="https://user-images.githubusercontent.com/15617021/103468784-03718280-4d5d-11eb-9fea-6d0ddab7ed19.png">

With that file the builds works fine.
But now we use `ios-auto-provision-appstoreconnect` which is working fine, too - but it destroys the `project.pbxproj` file.
Or it reformatting it so that Ionic / Cordova cannot read it anymore.

After the step the file looks like:

<img width="408" alt="2021-01-03 00_47_16- iOS - build #744 (d4436f25b3b693b3) - Bitrise" src="https://user-images.githubusercontent.com/15617021/103468797-4e8b9580-4d5d-11eb-894e-e04242a48c1a.png">

And `ionic-archive` is failing with this error message.

```
> cordova build ios --release --device --buildFlag=-UseModernBuildSystem=0
Could not parse /Users/vagrant/git/platforms/ios/App.xcodeproj/project.pbxproj: SyntaxError: Expected [\n\r] but end of input found.
[ERROR] An error occurred while running subprocess cordova.
```

I searched for the issue and someone has the idea to add a \n\r to the end of the file.
But even with that there is an issue:

```
> cordova build ios --release --device --buildFlag=-UseModernBuildSystem=0
Could not parse /Users/vagrant/git/platforms/ios/App.xcodeproj/project.pbxproj: TypeError: Cannot read property 'buildSettings' of undefined
[ERROR] An error occurred while running subprocess cordova.
```

**Build:** https://app.bitrise.io/build/d4436f25b3b693b3#?tab=log

**Comment by patrickbussmann** [comment]
I have a workaround.

Add `is_skippable` to `ionic-archive` step.

```
    - ionic-archive@2:
        is_skippable: true
```

And then add the auto provision and manual xcode archive after that step.

```
    - ios-auto-provision-appstoreconnect@0:
        inputs:
        - project_path: "$BITRISE_SOURCE_DIR/platforms/ios/App.xcworkspace"
        - scheme: App
        - distribution_type: "$BITRISE_EXPORT_METHOD"
    - xcode-archive@3:
        inputs:
        - project_path: "$BITRISE_SOURCE_DIR/platforms/ios/App.xcworkspace"
        - scheme: App
        - compile_bitcode: 'no'
        - upload_bitcode: 'no'
        - configuration: Release
        - export_method: "$BITRISE_EXPORT_METHOD"
```

Then it works.
So there must be something wrong with the cordova build.

**Comment by non-binary** [comment]
Thank you for reporting this. I've escalated it to the team responsible for looking into this.

**Comment by Robert-Bitrise** [comment]
Hi @patrickbussmann ,

Do you still notice this issue? Do you want to use the ionic-archive step in the future or currently you already have your own solution which is good enough?

I am looking forward to your reply.

**Comment by patrickbussmann** [comment]
Im still using the ionic-archive step.
But the first time is always failing.

I can work with my workaround but its still an issue.

**Comment by Robert-Bitrise** [comment]
Could you please create a test workflow to us to let us to be able to test? Thank you for your cooperation!

**Comment by Robert-Bitrise** [comment]
> Im still using the ionic-archive step.
> But the first time is always failing.
> 
> I can work with my workaround but its still an issue.

@patrickbussmann 
Could you please create a test workflow to us to let us to be able to test? Thank you for your cooperation!

**Comment by lpusok** [comment]
@patrickbussmann This may be caused by special characters in the Target names. Could you share the redacted names in the screenshot (https://user-images.githubusercontent.com/15617021/103468784-03718280-4d5d-11eb-9fea-6d0ddab7ed19.png) you shared?

**Comment by jpike88** [comment]
Ok, looks like I'm at this stage now. Happily awaiting a fix!

**Comment by jpike88** [comment]
Does any know of a workaround that doesn't involve ionic steps, and is entirely done in script calling Cordova directly? This is a blocking issue, we'll have to look for another product if there is not clear ETA on this issue.

**Comment by godrei** [comment]
Hi @jpike88 , sorry for the inconvenience!
We are working on this issue a few days ago.

Let me share where we are at the moment:

The issue is caused by a [3rd party library](https://github.com/DHowett/go-plist) we use for the Xcode project parsing and modification. 
This lib reformats the .pbxproj file by removing comments and changing indentation when we force the ios-auto-provison step defined codesign settings and save the project. These format changes are breaking the Cordova CLI.

We forked the library and started to fix this issue, you can see what we have already merged: https://github.com/bitrise-io/go-plist/pulls?q=is%3Apr+is%3Aclosed.

**Comment by adborbas** [comment]
Hi,

This issue should be resolved by the [1.0.3](https://github.com/bitrise-steplib/steps-ios-auto-provision-appstoreconnect/releases/tag/1.0.3) release.

Please feel free to open a new issue incase you still experience the problem.

Thanks
